clippy_lints/src/slice_iter_any.rs

```diff
@@ -0,0 +1,63 @@
+use crate::methods::method_call;
+use clippy_utils::diagnostics::span_lint;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::{self, IntTy, UintTy};
+use rustc_session::declare_lint_pass;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of `iter().any()` on slices of `u8` or `i8` and suggests using `contains()` instead.
+    ///
+    /// ### Why is this bad?
+    /// `iter().any()` on slices of `u8` or `i8` is optimized to use `memchr`.
```

Comment:
Don't you mean `.contains()`? Anyway, I think this is too specific, on my machine it seems to use compiler intrinsics and make no calls to `memchr`.

You're right about performances though: I benchmarked both `.iter().any(==)` and `.contains()` and the latter runs more than 10 times faster on large areas.

Reply:
Oh, I had assumed from [the implementation](https://github.com/rust-lang/rust/blob/4847d6a9d07d4be9ba3196f6ad444af2d7bdde72/library/core/src/slice/cmp.rs#L241-L259) that `memchr` would be used in any environment. Also, this gotbolt does so as far as the assembly is concerned: https://rust.godbolt.org/z/Kxrzr81Me

However, if it may differ depending on the environment, the description should be modified, so I'll make a change. Could you please tell me for reference, what's the environment you have checked that?

Reply:
I think the `memchr` itself has been replaced by an intrinsic. I'm using the latest nightly compiler on x86_64.

Reply:
Thank you! In any case, it looks like this lint description should be modified.

Reply:
I think the `memchr()` exists for larger integer types too, yes?

Reply:
oh, nope, it doesn't. It could be extended, I think.

Reply:
I found that `contains()` is now faster for certain types (`u16, u32, u64, i16, i32, i64, f32, f64, usize, isize`) compared to before (see: https://github.com/rust-lang/rust/pull/130991). Therefore, this performance lint should be extended to cover these types starting from Rust 1.84.0 and I'll make a change for this.